src/Subscriptions/SubscriptionRegistry.php

```diff
@@ -88,14 +98,24 @@ public function has(string $key): bool
      */
     public function keys(): array
     {
-        return array_keys($this->subscriptions);
+        return $this->subscriptionType === null
+            ? []
+            : $this->subscriptionType->getFieldNames();
     }
 
     /**
      * Get instance of subscription.
      */
     public function subscription(string $key): GraphQLSubscription
     {
+        if (isset($this->subscriptions[$key])) {
+            return $this->subscriptions[$key];
+        }
+
+        if ($this->subscriptionType !== null) {
+            $this->subscriptionType->getField($key);
```

Comment:
```suggestion
            // Loading the field has the side effect of triggering a call to 
            // \Nuwave\Lighthouse\Support\Contracts\ProvidesSubscriptionResolver::provideSubscriptionResolver()
            // which is then expected to call static::register().
            $this->subscriptionType->getField($key);
```

Sounds pretty convoluted when writing it down like this, perhaps we can simplify.

Reply:
Yes, I agree, but honestly I'm out of ideas without a huge overhaul where this whole registry is just removed at all... if you have a clean idea I'm all ears!